backend/plugin/advisor/mysql/advisor_column_set_default_for_not_null.go

```diff
@@ -180,11 +180,10 @@ func (*columnSetDefaultForNotNullChecker) needDefault(ctx mysql.IFieldDefinition
 	}
 
 	switch ctx.DataType().GetType_().GetTokenType() {
-	case mysql.MySQLParserBLOB_SYMBOL:
-		return false
-	case mysql.MySQLParserJSON_SYMBOL:
-		return false
-	case mysql.MySQLParserGEOMETRY_SYMBOL,
+	case mysql.MySQLParserBLOB_SYMBOL,
```

Comment:
Also derived data type, likes `LONGTEXT`，`MEDIUM TEXT`.
Ref: https://github.com/bytebase/bytebase/pull/9426/files#diff-bea9e80f5ab49c8beaaf15138b0a77106f9c4f355ed970f6c526ac4ac940ad3bR332-R347

Reply:
Add in test also